extensions/cdc-debezium/src/main/java/com/hazelcast/jet/cdc/impl/ChangeRecordImpl.java

```diff
@@ -194,10 +189,10 @@ public boolean equals(Object obj) {
             return false;
         }
         ChangeRecordImpl that = (ChangeRecordImpl) obj;
-        return this.sequenceSource == that.sequenceSource &&
-                this.sequenceValue == that.sequenceValue &&
-                this.keyJson.equals(that.keyJson) &&
-                Objects.equals(this.oldValue, that.oldValue) &&
-                Objects.equals(this.newValue, that.newValue);
+        return this.sequenceSource == that.sequenceSource
+                && this.sequenceValue == that.sequenceValue
+                && this.keyJson.equals(that.keyJson)
```

Comment:
I think it is better if Objects.equals() is used for keyJson, timestamp, database etc. fields too. Basically for all fields  that are not primitive.
Objects.equals() handles reference equality and null conditions

Reply:
a3087854fd3ffbfd5737c7022e3289b8d18c5c88 + 113e5f9db076157cb010ddbac375c07498a58da1